### ISSUE TITLE
Please include virtual/physical batch sizes in the tutorials

### ISSUE THREAD

**Issue by talcs** [pytorch/pytorch#84140]
### 📚 The doc issue

The tutorials in the website, e.g. the [CIFAR10 tutorial](https://pytorch.org/tutorials/beginner/blitz/cifar10_tutorial.html) only refer to the batch size when initializing the dataloader. The batch size used for the dataloader is the same batch size used for the optimization step:

```
batch_size = 4

trainset = torchvision.datasets.CIFAR10(root='./data', train=True,
                                        download=True, transform=transform)
trainloader = torch.utils.data.DataLoader(trainset, batch_size=batch_size,
                                          shuffle=True, num_workers=2)
```

However, when using a single GPU, people may want to simulate rather large virtual batch sizes, i.e., compute the gradients of many small "physical" batches, which are feasible on the GPU, but only take the optimization step when reaching the desired amount of samples in the "virtual" batch. It is not trivial, since it turns out that the [gradients of the small batches are summed](https://discuss.pytorch.org/t/how-are-losses-aggregated-over-multiple-computed-batches/160041). This means that the optimization step size taken using the final gradient will be K times larger than the user-specified learning rate, which may be undesirable in most cases. The losses of the small batches should therefore be multiplied by their relative size, so the final gradient will exactly simulate a large batch.

## Dataloader initialization
```
optimization_batch_size = 16
physical_batch_size = 4

trainset = torchvision.datasets.CIFAR10(root='./data', train=True,
                                        download=True, transform=transform)
trainloader = torch.utils.data.DataLoader(trainset, batch_size=optimization_batch_size,
                                          shuffle=True, num_workers=2)
```

## Training
```
running_loss = 0.0
for i, data in enumerate(trainloader, 0):
    # get the inputs; data is a list of [inputs, labels]
    inputs, labels = data
    
    # zero the parameter gradients
    optimizer.zero_grad()

    # run batch
    batch_loss = 0.0
    # compute all physical batches
    for j in range(0, len(inputs), physical_batch_size):
        end_index = min(j + physical_batch_size, len(inputs))
        inputs_j = inputs[j:end_index]
        labels_j = labels[j:end_index]
        # forward + backward
        outputs_j = net(inputs_j)
        loss = criterion(outputs_j, labels_j) * len(inputs_j) / float(len(inputs))
        batch_loss += loss.item()
        loss.backward()
    # optimize whole batch
    optimizer.step()

    # print statistics
    running_loss += batch_loss
    if i % 2000 == 1999:    # print every 2000 mini-batches
        print(f'[{epoch + 1}, {i + 1:5d}] loss: {running_loss / 2000:.3f}')
        running_loss = 0.0
```
